### ISSUE TITLE
ArgoCD should download helm chart dependencies

### ISSUE THREAD

**Issue by alexmt** [argoproj/argo-cd#582]
Helm allows specifying chart dependencies in requirements.yaml (example https://github.com/coreos/prometheus-operator/blob/master/helm/kube-prometheus/requirements.yaml ) . ArgoCD should (optionally?) download dependencies using `helm dependency update` in requirements.yaml exist.

**Comment by jessesuen** [comment]
This is tricky because we reset the repository for each checkout. Meaning we would need to download this dependency every time we want to perform a compare. That said, the fact that we cache manifests saves us from doing on every comparison... but it's still an expensive ask.

**Comment by jessesuen** [comment]
Turns out it's not that expensive since these are just tar.gz text files. Will perform `helm dependency update` as part of each manifest generation.

**Comment by jessesuen** [comment]
Fixed in v0.9

**Comment by zeph** [comment]
@jessesuen i'm not sure about the fix here... since it looks like if i don't declare as repo in the argocd each single repo which is in the requirements.yaml it won't work... and i cannot do that either cause I get back the following:

 ```
argocd repo add https://helm.elastic.co
FATA[0000] rpc error: code = Unimplemented desc = unknown method ValidateAccess
```
